### PR TITLE
APPS-3069: use in app webview

### PR DESCRIPTION
**Pull request by rostikjoystick**
**YouTrack task**: [#APPS-3069](https://vyahhi.myjetbrains.com/youtrack/issue/APPS-3069)

**Description List**:
* Open not supported quizzes inside in app web view
* Add "open in external browser" action to in app web view

### CODE REVIEW

File: app/src/main/res/values-be/strings.xml [hunk 1]
```diff
@@ -801,7 +801,9 @@
 
     <string name="filter_dismiss_action">Скінуць</string>
     <string name="filter_apply_action">Паглядзець вынік</string>
-    <!-- -->
+
+    <!-- In app web view (done via Google Translate) -->
+    <string name="in_app_web_view_step_title">Крок %d</string>
```
Comment (eadm) [inline app/src/main/res/values-be/strings.xml:806]: this str res should have step feature naming

File: app/src/main/java/org/stepik/android/view/in_app_web_view/InAppWebViewDialogFragment.kt [hunk 2]
```diff
@@ -116,6 +118,18 @@ class InAppWebViewDialogFragment : DialogFragment(), InAppWebViewView {
         centeredToolbarTitle.text = title
         centeredToolbar.setNavigationOnClickListener { dismiss() }
         centeredToolbar.setTintedNavigationIcon(R.drawable.ic_close_dark)
+        centeredToolbar.inflateMenu(R.menu.in_app_web_view_menu)
+        centeredToolbar.setOnMenuItemClickListener { menuItem ->
+            when (menuItem.itemId) {
+                R.id.menu_item_external -> {
+                    startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(url)))
+                    dismiss()
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/view/in_app_web_view/InAppWebViewDialogFragment.kt:126]: can it lead to state loss exception?

File: app/src/main/java/org/stepik/android/view/step_quiz/ui/fragment/DefaultStepQuizFragment.kt [hunk 3]
```diff
@@ -147,8 +147,8 @@ abstract class DefaultStepQuizFragment : Fragment(), StepQuizView {
     }
 
     private fun openStepInWeb(step: Step) {
-        MagicLinkDialogFragment
-            .newInstance(stepDeepLinkBuilder.createStepLink(step))
-            .showIfNotExists(childFragmentManager, MagicLinkDialogFragment.TAG)
+        InAppWebViewDialogFragment
+            .newInstance(getString(R.string.in_app_web_view_step_title, step.id), stepDeepLinkBuilder.createStepLink(step), isProvideAuth = true)
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/view/step_quiz/ui/fragment/DefaultStepQuizFragment.kt:151]: StepId doesn't mean anything to user, lets display lesson title instead

File: app/src/main/java/org/stepik/android/view/comment/ui/dialog/SolutionCommentDialogFragment.kt [hunk 4]
```diff
@@ -173,9 +173,9 @@ class SolutionCommentDialogFragment : DialogFragment() {
                         submissionDeepLinkBuilder.createSubmissionLink(step.id, submission.id)
                     }
 
-                MagicLinkDialogFragment
-                    .newInstance(url)
-                    .showIfNotExists(childFragmentManager, MagicLinkDialogFragment.TAG)
+                InAppWebViewDialogFragment
+                    .newInstance(getString(R.string.in_app_web_view_step_title, step.id), url, isProvideAuth = true)
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/view/comment/ui/dialog/SolutionCommentDialogFragment.kt:177]: title should be the same with one from SolutionCommentDialogFragment

File: app/src/main/java/org/stepik/android/view/in_app_web_view/InAppWebViewDialogFragment.kt (hunk 2 above)
Comment (rostikjoystick) [reply app/src/main/java/org/stepik/android/view/in_app_web_view/InAppWebViewDialogFragment.kt:126]: Similar to [this](https://github.com/StepicOrg/stepik-android/blob/a533c4180ec8d0f16320ac9e9a8c3b7ae5192187/app/src/main/java/org/stepik/android/view/magic_links/ui/dialog/MagicLinkDialogFragment.kt#L67), should be fine.

**Review by eadm — APPROVED** [review]

**Pull request merged**